lib/core/DocsLayout.js

```diff
@@ -35,16 +36,16 @@ class DocsLayout extends React.Component {
   render() {
     const metadata = this.props.metadata;
     const content = this.props.children;
-    const i18n = translation[this.props.metadata.language];
+    const i18n = translation[metadata.language];
+    const id = metadata.localized_id;
+    const defaultTitle = metadata.title;
     let DocComponent = Doc;
     if (this.props.Doc) {
       DocComponent = this.props.Doc;
     }
-    const title = i18n
-      ? translation[this.props.metadata.language]['localized-strings'][
-          this.props.metadata.localized_id
-        ] || this.props.metadata.title
-      : this.props.metadata.title;
+    const title =
+      utils.getDeepProp(i18n, `localized-strings.docs.${id}.title`) ||
```

Comment:
Good catch, will fix it!

Reply:
I am still in favor of using array. It is more declarative to me. The dot notation will work if there is the key does not contain a dot.

This is why we have to do `props['hello.world']` sometimes becayse `props.hello.world` wont work

Compare
`['localized-string', id]` vs ` 'localized-string.${id}'`

In fact just-safe-get library you mentioned also support accessing deep nested pros using array path

```js
import get from 'just-safe-get';
 
const obj = {a: {aa: {aaa: 2}}, b: 4};
 
get(obj, 'a.aa.aaa'); // 2
get(obj, ['a', 'aa', 'aaa']); // 2
```

Reply:
But we dont need a library for that. The `idx` function I mentioned in last review should be quite ok.

The naming of `idx` is because in Facebook we have that function. I think they even blogged about it before

https://facebook.github.io/react-native/blog/2017/03/13/idx-the-existential-function

Reply:
I fixed and renamed it back to idx!

Edited it a little bit to return actual value instead of returning `null` on every falsy value.